### ISSUE TITLE
Wrong Style/RedundantLineContinuation and autocorrect

### ISSUE THREAD

**Issue by jgraichen** [rubocop/rubocop#12727]
The Style/RedundantLineContinuation cop wrongly flags the following as a redundant line continuation since 1.61.0, _and_ applies a broken autocorrect:

```ruby
    expect(trace.meta).to eq \
      method: 'HEAD',
      path: '/',
      protocol: 'HTTP/1.1'
```

--------

## Expected behavior

RuboCop should not flag the line continuation nor remove it.

## Actual behavior

The linter falsy reports a redundant line continuation, and autocorrect breaks the file:

```console
> bundle exec rubocop -a --debug spec/mnemosyne/middleware/rack_spec.rb 
For ~/workspace/mnemosyne/ruby: configuration from ~/workspace/mnemosyne/ruby/.rubocop.yml
Default configuration from ~/.rvm/rubies/ruby-3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.61.0/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning ~/workspace/mnemosyne/ruby/spec/mnemosyne/middleware/rack_spec.rb
F

Offenses:

spec/mnemosyne/middleware/rack_spec.rb:57:30: C: [Corrected] Style/RedundantLineContinuation: Redundant line continuation.
    expect(trace.meta).to eq \ ...
                             ^
spec/mnemosyne/middleware/rack_spec.rb:58:13: F: Lint/Syntax: unexpected token tCOLON
(Using Ruby 2.7 parser; configure using TargetRubyVersion parameter, under AllCops)
      method: 'HEAD',
            ^
spec/mnemosyne/middleware/rack_spec.rb:82:37: C: [Corrected] Style/RedundantLineContinuation: Redundant line continuation.
      expect(trace.meta).to include \ ...
                                    ^

1 file inspected, 3 offenses detected, 2 offenses corrected
Finished in 0.09757550100039225 seconds
```

It applies the following change:

```diff
--- spec/mnemosyne/middleware/rack_spec.rb
+++ spec/mnemosyne/middleware/rack_spec.rb
@@ -54,7 +54,7 @@ RSpec.describe Mnemosyne::Middleware::Rack, probe: :rack do
     end
 
     expect(trace.name).to eq 'app.web.request.rack'
-    expect(trace.meta).to eq \
+    expect(trace.meta).to eq 
       method: 'HEAD',
       path: '/',
       protocol: 'HTTP/1.1',
@@ -79,7 +79,7 @@ RSpec.describe Mnemosyne::Middleware::Rack, probe: :rack do
       end
 
       expect(trace.name).to eq 'app.web.request.rack'
-      expect(trace.meta).to include \
+      expect(trace.meta).to include 
         location: 'http://www.example.org',
         status: 302
     end
```

## Steps to reproduce the problem

See this open-source pull request: https://github.com/mnemosyne-mon/mnemosyne-ruby/pull/75.

To reproduce, place the following files:

* `test.rb`

	```ruby
	# frozen_string_literal: true
	
	expect(trace.meta).to eq \
	  method: 'HEAD',
	  path: '/',
	  protocol: 'HTTP/1.1'
	```

* `.rubocop.yml`

	```
	AllCops:
	  NewCops: enable
	  SuggestExtensions: False
	```

Run rubocop:

```console
> rubocop --debug test.rb
For /tmp/tmp.jgEY9EI159: configuration from /tmp/tmp.jgEY9EI159/.rubocop.yml
Default configuration from /home/jan/.rvm/rubies/ruby-3.3.0/lib/ruby/gems/3.3.0/gems/rubocop-1.61.0/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /tmp/tmp.jgEY9EI159/test.rb
C

Offenses:

test.rb:3:26: C: [Correctable] Style/RedundantLineContinuation: Redundant line continuation.
expect(trace.meta).to eq \ ...
                         ^

1 file inspected, 1 offense detected, 1 offense autocorrectable
Finished in 0.1036660159998064 seconds
```

## RuboCop version

```
> bundle exec rubocop -V
1.61.0 (using Parser 3.3.0.5, rubocop-ast 1.31.0, running on ruby 3.2.2) [x86_64-linux]
```

**Comment by jgraichen** [comment]
Duplicate of #12726 posted at the same time. :grimacing:
